contracts/middle-layer/GreenfieldExecutor.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "../Config.sol";
+import "../interface/ICrossChain.sol";
+import "../interface/IGreenfieldExecutor.sol";
+
+contract GreenfieldExecutor is Config, Initializable, IGreenfieldExecutor {
+    enum MsgType {
+        Unspecified,
```

Comment:
I think we dont have to really list the support type in Enum, but just comment in this file. Because if the greenfield add one new kind of message, we dont want to upgrade this contract.

Reply:
Done